### PR TITLE
Upgrade pygments to v1.x

### PR DESCRIPTION
**Pull request by parkr**
This updates Pygments and our related tests to support Pygments.rb 1.x.

This is required to ship #5687.

/cc @jekyll/build @ashmaroli

### CODE REVIEW

**Comment by ashmaroli** [comment]
- Windows is incompatible with `pygments.rb 1.0 - 1.1.1` because of `symlinks`, and the pygments team do not seem to be keen on extending support to Windows.
- Ruby on Windows is still at `2.3.3`. Mentioning it here, as the changes proposed here was initiated by a proposal to use `Ruby 2.4.0`.

**Comment by parkr** [comment]
> Windows is incompatible with pygments.rb 1.0 - 1.1.1 because of symlinks, and the pygments team do not seem to be keen on extending support to Windows.

Ugh! The curse of getting developers to care about windows. Also, symlinks are a massively common and useful tool. Why does everything have to be so different??
> Ruby on Windows is still at 2.3.3. Mentioning it here, as the changes proposed here was initiated by a proposal to use Ruby 2.4.0.
Yeah, I just want to be ready for the upgrade. We also need this for Ruby 2.4 on Travis anyway.

**Comment by parkr** [comment]
To avoid the Windows issue with pygments.rb's upgrade to Pygments 2.2, I'm going to close this.

**Comment by ashmaroli** [comment]
TODO: Reconsider when Jekyll drops support for Ruby v2.2 and below.

**Comment by ashmaroli** [comment]
@parkr Apparently the folks at pygments.rb [removed the use of symlinks](https://github.com/tmm1/pygments.rb/commit/0ac219a748806b5e6523bbd112d17e47bc3effe7).
Which means this can be merged.. :tada: 

/cc @pup @DirtyF

File: test/test_redcarpet.rb [hunk 1]
```diff
@@ -46,16 +46,17 @@ class TestRedcarpet < JekyllUnitTest
...
       should "render fenced code blocks with syntax highlighting" do
-        assert_equal "<div class=\"highlight\"><pre><code class=\"language-ruby\" "\
-                     "data-lang=\"ruby\"><span class=\"nb\">puts</span> <span "\
-                     "class=\"s2\">&quot;Hello world&quot;</span>\n</code></pre></div>",
-                     @markdown.convert(
-                       <<-EOS
+        assert_equal \
+          "<div class=\"highlight\"><pre><code class=\"language-ruby\" " \
+          "data-lang=\"ruby\"><span></span><span class=\"nb\">puts</span> <span " \
+          "class=\"s2\">&quot;Hello world&quot;</span>\n</code></pre></div>",
```
Comment (ashmaroli) [inline test/test_redcarpet.rb]: We can ideally avoid escaping the double-quote here by using `%{}` like in `test_tags.rb`..
But on further inspection, there are multiple such instances.. best handled in a separate PR.. :+1:

Comment (parkr) [reply test/test_redcarpet.rb]: Feel free to take over this PR and commit directly to this branch.

**Comment by DirtyF** [comment]
Thanks @parkr ❤️ 

Let's avoid exposing our users to potential critical flaws. 
@jekyllbot: merge +dev

**Pull request merged**